experimental/dds/attributable-map/src/mapKernel.ts

```diff
@@ -318,6 +331,7 @@ export class MapKernel {
 	public delete(key: string): boolean {
 		// Delete the key locally first.
 		const previousValue = this.deleteCore(key, true);
+		this.setAttribution(key);
```

Comment:
IMO it makes more sense to clear attribution information here and in `deleteCore` than set it: a key that doesn't have a corresponding entry probably shouldn't have associated attribution information. You'll need to update the expectations in some of your tests.